### ISSUE TITLE
VaadinRequest getParameter not working

### ISSUE THREAD

**Issue by giusdiluca** [vaadin/flow#13114]
### Description of the bug

While trying an upgrade from Vaadin Flow 14.8.2 to Vaadin Flow 22.0.3 it seems that VaadinService.getCurrentRequest().getParameter("parameterName") is not working. This call returns null, while the same call is working correctly in Vaadin Flow 14.

### Expected behavior

VaadinRequest.getParameter("parameterName") should return the value of the parameter from the request

### Minimal reproducible example

String param = VaadinService.getCurrentRequest().getParameter("parameterName")
param == null

### Versions

- Vaadin / Flow version: 22.0.3
- Java version: 8
- OS version: RedHat 7.7
- Browser version (if applicable): Chrome 98
- Application Server (if applicable): Wildfly 14.0.1 Final
- IDE (if applicable): Netbeans 8.2

**Comment by Artur-** [comment]
Where do you call this? In your view code?

**Comment by knoobie** [comment]
Where are you using the method?

**Comment by giusdiluca** [comment]
> Where do you call this? In your view code?

I'm using it on the client side to retrieve authentication ticket from the URL

**Comment by caalador** [comment]
As the usage is not clear we would need a minimal project that can be used to see the issue.

**Comment by giusdiluca** [comment]
Hello, it's a bit difficult to give you a project, because the use case giving the issue is with regard an authentication of the Vaadin Application with CAS Authentication Server. We've noticed that the execution flow is slightly changed between Vaadin 14 and Vaadin 22. By deep diving the flow, we've seen that after the login the CAS parameters are in the request and then, after the session Init (VaadinServlet) the flow is passed correctly to an application class (Vaadin 14). This is different in Vaadin 22 where after the login the first request contains the CAS parameters expected but there is a multiple time filter chain traversal and also multiple VaadinServlet.service method invocation before the application class is invoked. 

So we've decided to save the parameters from the first request as session attributes, and that solved our issue.
